src/uwtools/drivers/mpas_init.py

```diff
@@ -0,0 +1,178 @@
+from uwtools.drivers.driver import Driver
+from uwtools.strings import STR
+from uwtools.utils.tasks import file, filecopy, symlink
+
+
+class MPASInit(Driver):
+    """
+    A driver for mpas-init.
+    """
+
+    def __init__(self, config: Path, cycle: datetime, dry_run: bool = False, batch: bool = False):
+        """
+        The driver.
+
+        :param config_file: Path to config file (read stdin if missing or None).
+        :param cycle: The cycle.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config=config, dry_run=dry_run, batch=batch, cycle=cycle)
+        if self._dry_run:
+            dryrun()
+        self._cycle = cycle
+
+    # Workflow tasks
+
+    @tasks
+    def boundary_files(self):
+        """
+        Boundary-condition files.
+        """
+        yield self._taskname("boundary files")
+        lbcs = self._driver_config["boundary_conditions"]
+        endhour = lbcs["length"]
+        interval = lbcs["interval_hours"]
+        symlinks = {}
+        boundary_filepath = lbcs["path"]
+        for boundary_hour in range(0, endhour + 1, interval):
+            file_date = self._cycle + timedelta(hours=boundary_hour)
+            target = Path(boundary_filepath) / f"FILE:{file_date.strftime('%Y-%m-%d_%H')}"
+            linkname = self._rundir / f"FILE:{file_date.strftime('%Y-%m-%d_%H')}"
+            symlinks[target] = linkname
+        yield [symlink(target=t, linkname=l) for t, l in symlinks.items()]
+
+    @tasks
+    def files_copied(self):
+        """
+        Files copied for run.
+        """
+        yield self._taskname("files copied")
+        yield [
+            filecopy(src=Path(src), dst=self._rundir / dst)
+            for dst, src in self._driver_config.get("files_to_copy", {}).items()
+        ]
+
+    @tasks
+    def files_linked(self):
+        """
+        Files linked for run.
+        """
+        yield self._taskname("files linked")
+        yield [
+            symlink(target=Path(target), linkname=self._rundir / linkname)
+            for linkname, target in self._driver_config.get("files_to_link", {}).items()
+        ]
+
+    @task
+    def namelist_file(self):
+        """
+        The namelist file.
+        """
+        fn = "namelist.init_atmosphere"
+        yield self._taskname(fn)
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        yield None
+        stop_time = self._cycle + timedelta(
+            hours=self._driver_config["boundary_conditions"]["length"]
+        )
+        d = {
+            "nhyd_model": {
+                "config_start_time": self._cycle.strftime("%Y-%m-%d_%H:00:00"),
+                "config_stop_time": stop_time.strftime("%Y-%m-%d_%H:00:00"),
+            }
+        }
+        namelist = self._driver_config.get("namelist", {})
+        values = YAMLConfig(d)
+        values.update_values(namelist.get("update_values", {}))
```

Comment:
Given this order of operations, it looks like we'd start with the known-correct cycle info in `d` and then potentially clobber that with possibly-wrong values taken from `update_values`. One would hope the user wouldn't provide bad cycle info in `update_values`, but I feel like the information provided by the `--cycle` switch (or equivalent API argument) should always take precedence. If that's right, how about something like
```
update_values = namelist.get("update_values", {})
update_values.setdefault("nhyd_model", {}).update({"config_start_time": self._cycle.strftime("%Y-%m-%d_%H:00:00"), "config_stop_time": stop_time.strftime("%Y-%m-%d_%H:00:00")})
```
(It'll look nicer when `black` wraps it.)

Here's a demo (I've set the `self._cycle` and `stop_time` variables):
```
>>> update_values = {}
>>> update_values.setdefault("nhyd_model", {}).update({"config_start_time": self._cycle.strftime("%Y-%m-%d_%H:00:00"), "config_stop_time": stop_time.strftime("%Y-%m-%d_%H:00:00")})
>>> update_values
{'nhyd_model': {'config_start_time': '2024-03-29_12:00:00', 'config_stop_time': '2024-03-30_12:00:00'}}
```
That's if `update_values` isn't provided in YAML, but it should work just as well if it is:
```
>>> update_values = {"foo": "bar", "nhyd_model": {"baz": "qux", "config_start_time": "bad-value", "config_stop_time": "bad-value"}}
>>> update_values.setdefault("nhyd_model", {}).update({"config_start_time": self._cycle.strftime("%Y-%m-%d_%H:00:00"), "config_stop_time": stop_time.strftime("%Y-%m-%d_%H:00:00")})
>>> update_values
{'foo': 'bar', 'nhyd_model': {'baz': 'qux', 'config_start_time': '2024-03-29_12:00:00', 'config_stop_time': '2024-03-30_12:00:00'}}
```
The `.setdefault()` call creates the `nhyd_model` `dict` if it doesn't exist yet, and otherwise just uses it.

Reply:
I will apply this suggested approach to the mpas driver.